test/sql.test.js

```diff
@@ -0,0 +1,307 @@
+var expect = require('chai').expect;
+var SqlConnector = require('../lib/sql');
+var ParameterizedSQL = SqlConnector.ParameterizedSQL;
+var testConnector = require('./connectors/test-sql-connector');
+
+var juggler = require('loopback-datasource-juggler');
+var ds = new juggler.DataSource({
+  connector: testConnector,
+  debug: true
+});
+var connector;
+
+describe('sql connector', function() {
+  before(function() {
+    connector = ds.connector;
+    connector._tables = {};
+    connector._models = {};
+    ds.createModel('customer', {
+        name: {
+          id: true,
+          type: String,
+          testdb: {
+            column: 'NAME',
+            dataType: 'VARCHAR',
+            dataLength: 32
+          }
+        }, vip: {
+          type: Boolean,
+          testdb: {
+            column: 'VIP'
+          }
+        },
+        address: String
+      },
+      {testdb: {table: 'CUSTOMER'}});
```

Comment:
The same comment applies to multiple other places in this file, I won't post comments for each one as that would spam the discussion here. If you agree with me that we should add spaces, then please look all places up yourself.